tests/models/whisper/test_modeling_whisper.py

```diff
@@ -1289,3 +1290,198 @@ def test_tiny_timestamp_generation(self):
 
         transcript = processor.batch_decode(generated_ids, skip_special_tokens=True, output_offsets=True)
         self.assertEqual(transcript, EXPECTED_TRANSCRIPT)
+
+
+def prepare_whisper_encoder_inputs_dict(
+    config,
+    input_features,
+    head_mask=None,
+):
+    if head_mask is None:
+        head_mask = torch.ones(config.encoder_layers, config.encoder_attention_heads, device=torch_device)
+    return {
+        "input_features": input_features,
+        "head_mask": head_mask,
+    }
+
+
+@require_torch
+class WhisperEncoderModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=13,
+        seq_length=60,
+        is_training=True,
+        use_labels=True,
+        hidden_size=16,
+        num_hidden_layers=2,
+        num_attention_heads=4,
+        input_channels=1,
+        hidden_act="gelu",
+        hidden_dropout_prob=0.1,
+        attention_probs_dropout_prob=0.1,
+        max_position_embeddings=20,
+        max_source_positions=30,
+        num_mel_bins=80,
+        num_conv_layers=1,
+        suppress_tokens=None,
+        begin_suppress_tokens=None,
+        classifier_proj_size=4,
+        num_labels=2,
+        is_encoder_decoder=False,
```

Comment:
The reason for the new model tester is that the existing model tester is designed for `WhisperForConditionalGeneration`, an encoder-decoder model that performs inference through generation.

We can either put `WhisperForAudioClassification` in the encoder-decoder model tester, and then add `if/else` statements to almost every test, skipping the decoder & generation logic as required, or add a new standalone model tester for the encoder-only model. IMO the latter of these two options is the cleanest, hence the new model tester!

Reply:
Sure! Thanks works for me 😉 I thought we could have like in NLP where we add a list of classes and test them all but usually they are all encoder decoders, so you are rigth